src/golang/lib/engine/aq_engine.go

```diff
@@ -780,33 +818,18 @@ func (eng *aqEngine) executeWithEngine(
 	dag dag_utils.WorkflowDag,
 	workflowName string,
 	engineConfig shared.EngineConfig,
-	storageConfig shared.StorageConfig,
 	workflowRunMetadata *WorkflowRunMetadata,
 	timeConfig *AqueductTimeConfig,
 	opExecMode operator.ExecutionMode,
 	vaultObject vault.Vault,
+	engJobManager *job.JobManager,
```

Comment:
Isn't `job.JobManager` an interface? So it's already a pointer - I'm confused as to why we need two levels of indirection, it seems unnecessary.